var/spack/repos/builtin/packages/masurca/package.py

```diff
@@ -24,6 +24,8 @@ class Masurca(Package):
     patch('arm.patch', when='target=aarch64:')
 
     def patch(self):
+        filter_file('#include <sys/sysctl.h>', '',
+                    'global-1/CA8/src/AS_BAT/memoryMappedFile.H')
         if self.spec.target.family == 'aarch64':
```

Comment:
```suggestion
    def patch(self):
        if self.spec.target.family == 'aarch64':
```
By removing the #include, you remove the declaration, and that would turn the function call into a function call without a prototype. Maybe glibc just removed the exported header but still has the function, or it is still just a syscall wrapper in glibc.

What you should likely do rather is to contact the devs (likey thru their github commit accounts) and make them aware of this and add a patch, likely using the alternative code path in #else below the call to it instead, and then
you can update to their new version and use the same upstream patch instead: https://github.com/alekseyzimin/masurca

Activating the alternative code should be as simple as ensuring that that define is not #defines. This is likely done in the configure script to equivalent:
```
#ifdef HW_PHYSMEM
```

Reply:
I have a PR open on the submodule in question. https://github.com/alekseyzimin/wgs-8/pull/1 It hasn't had any traffic yet.

HW_PHYSMEM is not defined by anything in the build chain. It strikes me as a holdover from the past that was only ever manually activated. 

My PR does the more correct thing of adding a guard around the include. It was just easier to remove the include in the recipe to make it build. When a new version is cut with the fix in we can just add a version restriction to the filter_file.

Reply:
Thanks, since you mentioned https://github.com/alekseyzimin/wgs-8/pull/1 and explained why it can be removed in this comment, I hope the upstream developers might see and understand your PR too. It's clear to me now.

PS: You could actually insert newlines with `filter_file()` using `\n`, so it would have been possible to make the same change as in your upstream PR with it, but I agree that it's not required in this case for `spack`.

To explain the removal as briefly as possible, I'd squash the git commit message for the merge like this:
```
Subject: masurca: Fix build with glibc-2.32+

remove sysctl.h which is not used by the build
```